From 02d1eeef7df0cff4db1c46f2cb0125a03c173c7d Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Wed, 8 Oct 2014 15:54:10 +0200
Subject: [PATCH 4/4] bcmdhd: disable channel cache roaming

Currently (driver 1.201 + K3.14), this
feature is preventing basic Wifi features to work
(connect to an Access Point is not possible).
Also, fix the code when this flag is not defined.

Change-Id: Iaff8465c19b68ef2aa26ee1b542127a46982c113
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 drivers/net/wireless/bcmdhd/Makefile      |    3 ++-
 drivers/net/wireless/bcmdhd/wl_cfg80211.c |    2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index 7f2acf2833ce..0a393b079e7b 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -38,8 +38,9 @@ DHDCFLAGS += -DCUSTOM_TDLS_RSSI_THRESHOLD_HIGH=-60
 DHDCFLAGS += -DCUSTOM_TDLS_RSSI_THRESHOLD_LOW=-70
 
 # Roaming
+#DHDCFLAGS += -DROAM_CHANNEL_CACHE
 DHDCFLAGS += -DROAM_AP_ENV_DETECTION
-DHDCFLAGS += -DROAM_ENABLE -DROAM_CHANNEL_CACHE -DROAM_API
+DHDCFLAGS += -DROAM_ENABLE -DROAM_API
 DHDCFLAGS += -DENABLE_FW_ROAM_SUSPEND
 # Roaming trigger
 DHDCFLAGS += -DCUSTOM_ROAM_TRIGGER_SETTING=-75
diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index 859b8de06b59..acb12471ffd1 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -4027,8 +4027,10 @@ wl_cfg80211_connect(struct wiphy *wiphy, struct net_device *dev,
 	ext_join_params->assoc.chanspec_num = chan_cnt;
 	if (chan_cnt) {
 		if (use_chan_cache) {
+#ifdef ROAM_CHANNEL_CACHE
 			memcpy(ext_join_params->assoc.chanspec_list, chanspec_list,
 				sizeof(chanspec_t) * chan_cnt);
+#endif
 		} else {
 			u16 channel, band, bw, ctl_sb;
 			chanspec_t chspec;
-- 
1.7.9.5

